tools/api-errors.go

```diff
@@ -0,0 +1,49 @@
+package tools
+
+import (
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+type RequestError struct {
+	Status        int
+	CustomMessage string
+	Err           error
+}
+
+func (r RequestError) Error() string {
+	if r.Err != nil {
+		return r.Err.Error()
+	} else {
+		return ""
+	}
+}
+
+func (r RequestError) ToResponse() gin.H {
+	res := gin.H{"status": r.Status, "message": r.CustomMessage}
+
+	if r.Err != nil {
+		res["error"] = r.Error()
+	}
+
+	return res
+}
+
+func ErrorHandler(c *gin.Context) {
+	c.Next()
+
+	if len(c.Errors) > 0 {
```

Comment:
IMO for good measure this middleware should also abort the context if there are errors, so the request is not passed to other potential middlewares or handlers afterward.

Reply:
Added `c.Abort()` if an error occurs 👍🏻